harness/determined/cli/token.py

```diff
@@ -85,8 +85,11 @@ def create_token(args: argparse.Namespace) -> None:
             raise errors.CliError(f"User '{username}' not found or does not have an ID")
 
         # convert days into hours Go duration format
-        if args.expiration_days:
-            expiration_in_hours = str(24 * args.expiration_days) + "h"
+        expiration_in_hours = None
+        if args.expiration_days is not None:
+            expiration_in_hours = (
+                "-1" if args.expiration_days == -1 else f"{24 * args.expiration_days}h"
```

Comment:
might be worth just making sure that args.expiration_days is an int here, if it's a float not sure what would happen

Reply:
With the current code, any float input would return an error `invalid int value: '5.0'`. **I think we should keep it as is.**

Another option is to explicitly cast the input to an integer. However, the potential issue with this approach is that it could mislead users into thinking that floating-point values are valid, which might give the impression that fractional days are allowed. 

```
expiration_in_hours = None
if args.expiration_days is not None:
    expiration_days = int(args.expiration_days)  # Ensure it's an integer
    expiration_in_hours = "-1" if expiration_days == -1 else f"{24 * expiration_days}h"
```